app/src/resolvers.ts

```diff
@@ -8,31 +8,49 @@ import { getJenkinsServerWithSecret } from './storage/get-jenkins-server-with-se
 import { updateJenkinsServer } from './storage/update-jenkins-server';
 import { deleteBuilds } from './jira-client/delete-builds';
 import { deleteDeployments } from './jira-client/delete-deployments';
+import { isAdmin } from './check-permissions';
 
 const resolver = new Resolver();
 
-resolver.define('fetchJenkinsEventHandlerUrl', async () => ({
-	url: await webTrigger.getUrl('jenkins-webtrigger')
-}));
+const checkPermissions = async (accountId: string) => {
+	const admin = await isAdmin(accountId);
+	if (!admin) {
+		throw new Error('Only Jira administrators can perform this operation.');
+	}
+};
+
+resolver.define('fetchJenkinsEventHandlerUrl', async (req) => {
+	await checkPermissions(req.context.accountId);
+	return {
+		url: await webTrigger.getUrl('jenkins-webtrigger')
+	};
+});
 
 resolver.define('connectJenkinsServer', async (req) => {
+	await checkPermissions(req.context.accountId);
```

Comment:
can the resolver do multiple functions or only one?  If they can do multiple, just add `checkPermissions` before the inline function.  Another acceptable pattern is to wrap the inline function with the `checkPermissions` which in turn triggers the callback.

Reply:
I also feel that if forge can't handle authorization, there's a bigger issue here...

Reply:
That's a good point. I can fail from within the function to remove the `checkPermissions` func. However, I was concerned it may not be clear what it returns (error) if it doesn't go through, which is why I opted to have it at the top of the file. Let's discuss on Monday to see if we can come up with a readable option for this..

Reply:
Resolver can only have one func and doesn't support multiple callbacks. 

Discussed on zoom the advantages of this format (wrapping the func) but decided the readability was poor and stuck to the current format
```
resolver.define('getJenkinsServerWithSecret', adminPermissionCheck(async (req) => {
	const jenkinsServerUuid = req.payload.uuid as string;
	return getJenkinsServerWithSecret(jenkinsServerUuid);
}));
```